Add xmllint to prerequisites check (phase 01)

## Summary

The `generate-summary.sh` script requires `xmllint` (from `libxml2-utils`) to parse JUnit XML test results, but it is not checked in the prerequisites phase (01_check_dependencies_test.go).

When `xmllint` is missing, all tests pass but `make test-all` fails at the summary generation step, causing the entire CI run to report as failed:

```
=== All Test Phases Completed Successfully ===

Error: xmllint is required but not installed
make[1]: *** [Makefile:323: _test-all-impl] Error 1
```

Seen in: https://github.com/stolostron/capi-tests/actions/runs/22798133065/job/66135711173#step:12:2285

## What needs to happen

- Add a `TestCheckDependencies_XmllintAvailable` test in `test/01_check_dependencies_test.go` to check for `xmllint` availability
- Follow the existing pattern used for other tool checks (e.g., `kubectl`, `helm`, `kind`)
- Also install `libxml2-utils` in GHA workflows that run `make test-all`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/.github/workflows/workload-cluster-aro.yml
+++ b/.github/workflows/workload-cluster-aro.yml
@@ -87,7 +87,9 @@ jobs:
         cache: true
 
     - name: Install dependencies
-      run: go mod download
+      run: |
+        sudo apt-get update && sudo apt-get install -y libxml2-utils
+        go mod download
 
     - name: Install OpenShift CLI
       run: |

--- a/.github/workflows/workload-cluster-rosa.yml
+++ b/.github/workflows/workload-cluster-rosa.yml
@@ -91,7 +91,9 @@ jobs:
         cache: true
 
     - name: Install dependencies
-      run: go mod download
+      run: |
+        sudo apt-get update && sudo apt-get install -y libxml2-utils
+        go mod download
 
     - name: Install OpenShift CLI
       run: |

--- a/test/01_check_dependencies_test.go
+++ b/test/01_check_dependencies_test.go
@@ -23,6 +23,7 @@ func TestCheckDependencies_ToolAvailable(t *testing.T) {
 		"git",
 		"kubectl",
 		"go",
+		"xmllint",
 	}
 
 	// Add provider-specific tools (e.g., "az" for ARO, "aws" for ROSA)
@@ -896,6 +897,10 @@ func getToolInstallInstructions(tool string) string {
 			"  macOS: brew install go\n" +
 			"  Linux: Download from https://go.dev/dl/ and extract to /usr/local\n" +
 			"  All: https://go.dev/doc/install",
+		"xmllint": "Install xmllint (used by generate-summary.sh for JUnit XML parsing):\n" +
+			"  macOS: Pre-installed (part of libxml2)\n" +
+			"  Debian/Ubuntu: sudo apt-get install libxml2-utils\n" +
+			"  Fedora/RHEL: sudo dnf install libxml2",
 		"aws": "Install AWS CLI:\n" +
 			"  macOS: brew install awscli\n" +
 			"  Linux: curl \"https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip\" -o \"awscliv2.zip\" && unzip awscliv2.zip && sudo ./aws/install\n" +

